src/pages/index.js

```diff
@@ -49,10 +110,62 @@ constants.addNewCardButton.addEventListener("click", () => {
 });
 
 function handleImageSubmit(card) {
-  renderCard(card);
-  addNewCardModal.close();
-  constants.imageSubmitForm.reset();
-  addImageFormValidator.resetValidation();
+  addNewCardModal.setLoading(true, "Saving...");
+  api
+    .addCard(card)
+    .then(() => {
+      renderCard(card);
+    })
+    .then(() => {
+      addNewCardModal.setLoading(false);
+      constants.imageSubmitForm.reset();
+      addImageFormValidator.resetValidation();
+      addNewCardModal.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+}
+
+//Card like && dislike
+
+function likeButton(card) {
+  if (card.isLiked()) {
+    api
+      .dislikeCard(card._id)
+      .then((response) => card.setIsLiked(response.isLiked))
+      .catch((err) => console.error(err));
+  } else {
+    api
+      .likeCard(card._id)
+      .then((response) => card.setIsLiked(response.isLiked))
+      .catch((err) => console.error(err));
+  }
+}
+
+//Delete card confirmation modal
+
+const confirmationModal = new ModalWithConfirmation(
+  "#verification-modal",
+  handleDeleteCardForm
+);
+confirmationModal.setEventListeners();
+
+function handleDeleteCardForm(id, element) {
+  confirmationModal.open();
+  constants.deleteConfirmForm.addEventListener("submit", () => {
+    confirmationModal.setLoading(true, "Deleting...");
+    api
+      .deleteCard(id)
+      .then(() => element.remove())
+      .then(() => {
+        confirmationModal.setLoading(false);
+        confirmationModal.close();
```

Comment:
Needs correction: please move ` confirmationModal.close();` to `then` block, which is located on line 160